xilinx/src/xlnx_decoder.rs

```diff
@@ -0,0 +1,288 @@
+use crate::sys::*;
+use crate::xlnx_dec_utils::*;
+use crate::XlnxError;
+use simple_error::{bail, SimpleError};
+
+pub struct XlnxDecoder {
+    pub dec_session: *mut XmaDecoderSession,
+    pub frame_props: XmaFrameProperties,
+    in_buf: XmaDataBuffer,
+    pub out_frame: *mut XmaFrame,
+    pub flush_sent: bool,
+}
+
+impl XlnxDecoder {
+    pub fn new(xma_dec_props: &mut XmaDecoderProperties, xlnx_dec_ctx: &mut XlnxDecoderXrmCtx) -> Result<Self, SimpleError> {
+        // reserve the required decoding resources and assign reserve ID
+        xlnx_reserve_dec_resource(xlnx_dec_ctx)?;
+
+        let dec_session = xlnx_create_dec_session(xma_dec_props, xlnx_dec_ctx)?;
+
+        let mut frame_props: XmaFrameProperties = Default::default();
+        let ret = unsafe { xma_dec_session_get_properties(dec_session, &mut frame_props) };
+        if ret != XMA_SUCCESS as i32 {
+            bail!("unable to get frame properties from decoder session")
+        }
+
+        let in_buf: XmaDataBuffer = Default::default();//unsafe { xma_data_buffer_alloc(0, true) };
```

Comment:
Is this commented out code needed?

```suggestion
        let in_buf: XmaDataBuffer = Default::default();
```